### ISSUE TITLE
Improvements for Recap References for Year-in-review newsletter formats

### ISSUE THREAD

**Issue by bitschmidty** [bitcoinops/bitcoinops.github.io#1198]
I should have tested for the year in review formatting when the original recap auto-references PR was opened, my bad 🙋 

In creating the 231 year-in-review recap, I ran into some issues. It seems the featured summaries sections are not handled correctly by the recap references generator

see PR here: https://github.com/bitcoinops/bitcoinops.github.io/pull/1197
a local preview here: 
<img width="1355" alt="Screen Shot 2023-07-12 at 10 23 50 AM" src="https://github.com/bitcoinops/bitcoinops.github.io/assets/1615772/92e2bead-e04a-4908-add1-6df000888c8c">

- improvement to support handling the "featured summaries" in year in review (https://bitcoinops.org/en/newsletters/2022/12/21/)
- nice to have: linking "within" a month for the subtopics in that month. Although I think the month-only timestamps is fine

**Comment by harding** [comment]
Obviously, please feel free to make any changes to the year-in-review formatting to accommodate the recap stuff.  For example, to encapsulate stuff in HTML5 `<section>` tags or to use a hidden header like:

```
{:.hidden}
### Fee sponsorship
```

**Comment by kouloumos** [comment]
I'll get to this soon.

**Comment by bitschmidty** [comment]
@kouloumos I figured out a few formatting tweaks that work with the current architecture for YiR 2023, news/pod282.

https://github.com/bitcoinops/bitcoinops.github.io/pull/1449

So I think we are good here. I just need to apply those to YiR 2022, news/pod231
